rts/idris_utf8.c

```diff
@@ -78,6 +78,36 @@ unsigned idris_utf8_index(char* s, int idx) {
    return top;
 }
 
+char* idris_utf8_advance(char* str, int i) {
```

Comment:
@mbj please don't be sorry - I explicitly asked for help making this correct, because C-level bit-munging isn't the thing in the world that I'm best at. I appreciate it!